### ISSUE TITLE
Validation and Accessibility Issues on Homepage

### ISSUE THREAD

**Issue by sinabahram** [atbcb/usab-uswds#808]
This probably should be broken up into several child issues and assigned, then itself as a large issue closed. I leave that to the appropriate folks. Also, thanks for all the work on the site. The below is short/to-the-point out of respect for your time, not in order to be discourteous.

1. The homepage has several validation issues that can be found at:
http://validator.w3.org/nu/?doc=https%3A%2F%2Fwww.access-board.gov%2F%23content

The warnings about type on script tags and complaining about integrity are obviously low priority, but the other issues are directly or tangentially related to accessibility e.g. the multiple main regions on the page discussed below.

2. There are multiple main regions

3. The skip to content link does not seem to work in Chrome (have not tested elsewhere)

4. There are no h1's or h2's but h3's on the page, so if some invisible headings could be added to enhance the hierarchical organization of information, that would be most appreciated and would not incur a visual change at this time (larger discussion may be recommended on that front, though). H2's to preface the navigation and footer would be great (those can be offscreen).

5. The alt text repeatedly says "photo of". This is distracting and unnecessary given the context of the photos. Yes, sometimes medium should be specified, but not in this case with stock photos and such. It would really clean up the page to remove "photo of" on the alt text.

6. The order of your images and headings are swapped. This is super-common, but never-the-less pretty undiscoverable. You have obviously put in work to decorate these images with alt text, but most folks will never even encounter it because the images appear before the headings in the dom. Good news is they are not interactive, so you can swap them in the DOM, but preserve your visual order. That would not be true if they had both been links of course as it would have violated a visual expectation of tab traversal order, so can those please be swapped?

7. Accessibility is misspelled in this link "ICT Accessibilty Standards issued under Section 508 of the Rehabilitation Act and Section 255 of the Communications Act."

There are some other issues, but I just wanted to let you know of the above since they all appear on the homepage.

**Comment by bruce-usab** [comment]
Thanks @sinabahram for commenting!

1.  The main error is about the integrity attribute being used on images.  My understanding is that the CDN uses this value to facilitate caching.  I expect HTML5 to catch up.

2.  Addressed by PR #806. 

3.  The Skipnav on the homepage was broken.  Skipnav should be working in Chrome throughout site, as that is the primary browser we test with.  Please double check that!

4.  Noted.  We make extensive use of Hs with our longer documents.  I am not sure how soon we might revisit the overall structure.

5.  I am happy to discuss this, but I am of the school of thought that the ALT should be short but not so short as to preclude a screen reader user discussing the page with a sighted user.  Hence we are using "photo of office building" instead of just "building" or even null (since it is decorative).

6.  We might revisit the homepage structure.  We also do not like how the heading and images are not links, only the description text.  Again, this might be something we address later rather than sooner.

7.  Yikes!  PR #812.  Thanks for that!

**Comment by sinabahram** [comment]
Thanks for these responses.

In an incognito tab, the skip link issue appears to be resolved now.

RE alt text, length is not my concern there e.g. even a very brief description of that office building would actually make things more inclusive and more aesthetically accessible. My issue concerned the use of "photo of" preceding every single image. That tends to violate the principle of semantic prioritization e.g. we don't say "button ok, button cancel, button apply", but instead we say "OK button, Cancel button, Apply button" when discussing eyes-free interface design. Similarly here, I'm saying that you should offer a brief description of the photos. I  think they add value, but is it possible to please update your description guidelines to bring them inline with some best practices about stating media first when the media is 90% of the time the same thing?

So just to be super-clear, I'm not at all advocating for shorter alt text, in fact the opposite in some cases where it can add value, but I am advocating for making this alt text far better by removing a common phrase from the beginning, thereby reducing aural noise. Please let me know if that makes sense.

RE images and headings, you can decouple the conversation of making them both links from simply swapping their DOM order. Making them both links is doable, but I strongly advocate for an onClick, not a link, on the image, with a link on the heading, so as to avoid double-tab-stops, but still ensure keyboard accessibility (happy to help you with that pattern as we have implemented it dozens of times). But, for now, is it just possible to swap the order while keeping the visual the same e.g. flexbox, etc.?

Thanks for the rapid responses to both this and email feedback. Huge ratitude about that.

**Comment by bruce-usab** [comment]
I have bettered the alt text on the homepage.  I definitely should know better than to start all the alts with same word, shame on me!

Can you point me to a resource on swapping the DOM order?  I am not familiar with that other than swapping the source code order, which is not really an option for the template we built from.

I am also intrigued by your suggestion to add onClick to avoid double-tab-stops.  I am going to talk with @kengdoj about that one!

**Comment by sinabahram** [comment]
Ahh, without getting a bit complex via aria-owns or other techniques, switching the source code order is what I'm referring to for switching the DOM order, so it sounds like that may not be easily possible in the near future. Hopefully that can be baked into some upstream patch to the template to help more folks down the road, though.

**Comment by bruce-usab** [comment]
I meant to say so before, but I have huge gratitude for the several many people taking the time to provide feedback!  Thanks very much!
